### PR TITLE
Update hypershift and CS image

### PR DESCRIPTION
**Pull request by ahitacat**
### What this PR does
Updates the hypershift-operator and CS image to have the latest changes of hypershift.

Jira: 
Link to demo recording:

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -220,4 +220,4 @@ clouds:
           # Hypershift
           # uncomment the following line if you want to install the hypershift operator
           # with CRD support for managedIdentities
-          #hypershiftAdditionalInstallArg: '--tech-preview-no-upgrade'
+          hypershiftAdditionalInstallArg: '--tech-preview-no-upgrade'
```
Comment (ahitacat) [inline config/config.yaml:223]: This is required to create clusters with the latest hypershift-operator. 
I'm not sure if that will be needed in the rest of env (dev and cs-pr).

Comment (janboll) [reply config/config.yaml:223]: Please change to:
```yaml
hypershift:
   additionalInstallArg: "--tech-preview-no-upgrade"
```

**Comment by janboll** [comment]
You also need to run `make materialize` in the config directory as well

**Review by janboll — APPROVED** [review]

**Comment by machi1990** [comment]
Since we are bumping CS,
I'd like to point out that this change requires a change of the provision shard config.
Similar to what is done here https://github.com/Azure/ARO-HCP/pull/819 
Additionally, as that change is going to impact how we interact with the provision shard, it'll also require the dev env to be recreated as it is a breaking change - the current CS commit doesn't contain a migrations script

**Comment by geoberle** [comment]
Do we need to recreate the dev env also when there are no active clusters in the DB?

**Comment by geoberle** [comment]
Ah just read the no-mig-script comment

**Pull request merged**